tests/sqlparser_snowflake.rs

```diff
@@ -2256,3 +2256,9 @@ fn asof_joins() {
           "ORDER BY s.observed",
     ));
 }
+
+#[test]
+fn test() {
+    snowflake().verified_query("SELECT position('an', 'banana', 1)");
+    snowflake().verified_query("SELECT n, h, POSITION(n IN h) FROM pos");
```

Comment:
We can probably include an assertion demonstrating that the position parse to different expression types?

Also would it make more sense to have this in common given the code isn't guarded by snowflake dialect?

Reply:
Done